packages/api/endpoints/granule-csv.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+
+const { Parser } = require('json2csv');
+const router = require('express-promise-router')();
+
+const { Granule } = require('../models');
+
+/**
+ * Builds a CSV file of all granules in the Cumulus DB
+ *
+ * @param {Object} req - express request object
+ * @param {Object} res - express response object
+ * @returns {Object} the csv file of granules
+ */
+async function list(req, res) {
+  const granuleModel = new Granule();
+  const allDbGranules = await granuleModel.scan();
```

Comment:
Does the scan return all the granules?  If there are millions of granules, the memory could be any issue.  Maybe you could use the `DynamoDbSearchQueue`, and also search only the fields you need to reduce the memory.

Reply:
Alright I updated the scan to use the DynamoDbSearchQueue. Can you take a look? Is this what you had in mind?